.github/workflows/ci.yml

```
@@ -0,0 +1,45 @@
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[dev]
+
+      - name: Run unit tests with pytest
+        run: |
+          pytest tests/ --cov=rdd --cov-report=xml --cov-report=term
+
+      - name: Upload coverage report to Codecov (without token)
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.8)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)